File: src/main/java/teampearl/cvgenerator/back_end/presentation_layer/dto/TrainingParticipationDTO.java
package teampearl.cvgenerator.back_end.presentation_layer.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import teampearl.cvgenerator.back_end.presentation_layer.exceptions.DateIsInFutureException;

import java.time.LocalDate;

public class TrainingParticipationDTO extends Participation{
    @JsonProperty("id")
    private Integer id;

    @JsonProperty("employee")
    private EmployeeDTO employee;

    @JsonProperty("training")
    private TrainingDTO training;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public EmployeeDTO getEmployee() {
        return employee;
    }

    public void setEmployee(EmployeeDTO employee) {
        this.employee = employee;
    }

    public TrainingDTO getTraining() {
        return training;
    }

    public void setTraining(TrainingDTO training) {
        this.training = training;
    }

}

File: src/app/dto/TrainingParticipationDTO.ts
import { TrainingDTO } from './TrainingDTO';
import { EmployeeDTO } from './EmployeeDTO';
import { DTO } from './DTO';

export class TrainingParticipationDTO implements DTO{
  id!: number;
  employee!: EmployeeDTO;
  training!: TrainingDTO;
  beginDate!: Date;
  endDate!: Date;

  constructor() {
    this.employee = new EmployeeDTO;
    this.training = new TrainingDTO;

    this.beginDate = new Date;
    this.endDate = new Date;
  }
}
